### DISCUSSION TITLE
Use a job-level conditional/if to check command output

### DISCUSSION THREAD

**Discussion by mrcnski** [community/community#25081] [Actions]
<p>Hi, I want to skip a job if the last commit message started with a certain string.</p>
<p>The best we could come up with so far is to have the following as the first step:</p>
<pre><code># Is this a version change commit?
      - shell: bash
        id: commit_message
        run: |
          commit_message=$(git log --format=%B -n 1 ${{ github.sha }})
          echo "::set-output name=commit_message::$commit_message"
</code></pre>
<p>After which we include this condition on every following step:</p>
<pre><code>if: startsWith(steps.commit_message.outputs.commit_message, 'Version change')
</code></pre>
<p>However, this is verbose and prone to error if we forget the if on any subsequent step.</p>
<p>I’ve tried using a job-level conditional to check the output of a command or script, but it doesn’t seem possible within an expression.</p>
<p>Maybe someone has other ideas on how to accomplish what I’m trying to do?</p>
<p>Thanks!</p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p><strong>Output</strong> variable is inside a job. You could not use it in job-level condition.</p>
<p>Which trigger event do you use? If you are using <strong>on: push</strong> , you could get the last commit message in github.event context.</p>
<p>You could use next two conditions to filter commit message:</p>
<pre><code>if:  startsWith( github.event.commits[0].message , 'Version change')  
if:  startsWith( github.event.head_commit.message , 'Version change')
</code></pre>
<p>Please see my example:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/d/d1ac3d914de730db0ae6ff8567f9ae36e6b5d8f4.png" data-download-href="/uploads/short-url/tUQOwr3owIQMmYTa7QynNUs7zla.png?dl=1" title="commit message.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181088684-a787d84b-837c-4a3e-be16-68fc2cdc89c5.png" alt="commit message.png" data-base62-sha1="tUQOwr3owIQMmYTa7QynNUs7zla" width="690" height="332" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/d/d1ac3d914de730db0ae6ff8567f9ae36e6b5d8f4_2_690x332.png, https://aws1.discourse-cdn.com/github/original/2X/d/d1ac3d914de730db0ae6ff8567f9ae36e6b5d8f4.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/d/d1ac3d914de730db0ae6ff8567f9ae36e6b5d8f4.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/d/d1ac3d914de730db0ae6ff8567f9ae36e6b5d8f4_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">commit message.png</span><span class="informations">901×434 18.5 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by mrcnski** [comment]
<p>Thanks!</p>
<p>If I use the last commit message in several different jobs, is it possible to set it in a top-level env and reference it in an expression? I tried accessing it via</p>
<pre><code>env.COMMIT_MESSAGE
</code></pre>
<p>as well as</p>
<pre><code>github.env.COMMIT_MESSAGE
</code></pre>
<p>but neither worked. Anyway, this is just for a minor reduction in verbosity. Thanks again!</p>
